Bottom: c5c674f5b5fbda69d69ae9879857ac6766ea94f0
Top:    ad737718a5de9a9b003a3da018f631539f0a025e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 05:40:54 +0100

Include glib.h from global.h

To avoid compiler warnings, remove the now duplicated (or unecessary)
#define ABS definitions from various files.

There has been a build-time dependance on GLib since the toporouter was
added. Lets start to make use of it where it is useful.


---

diff --git a/src/autoroute.c b/src/autoroute.c
index c17a8fe..b58600f 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -159,10 +159,6 @@ static hidGC ar_gc = 0;
 	 CONFLICT_LEVEL(rb)==LO_CONFLICT ? \
 	 AutoRouteParameters.LastConflictPenalty : 1) * (rb)->pass)
 
-#if !defined(ABS)
-#define ABS(x) (((x)<0)?-(x):(x))
-#endif
-
 #define _NORTH 1
 #define _EAST 2
 #define _SOUTH 4
diff --git a/src/crosshair.c b/src/crosshair.c
index 2abe960..c2551ec 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -56,10 +56,6 @@
 
 RCSID ("$Id$");
 
-#if !defined(ABS)
-#define ABS(x) (((x)<0)?-(x):(x))
-#endif
-
 typedef struct
 {
   int x, y;
diff --git a/src/edif.y b/src/edif.y
index da33c9b..cb31d9a 100644
--- a/src/edif.y
+++ b/src/edif.y
@@ -38,7 +38,6 @@ LibraryEntryTypePtr GetLibraryEntryMemory (LibraryMenuTypePtr);
 #define	IDENT_LENGTH		255
 #define	Malloc(s)		malloc(s)
 #define	Free(p)			free(p)
-#define	ABS(v)			((v) < 0 ? -(v) : (v))
 #define	Getc(s)			getc(s)
 #define	Ungetc(c)		ungetc(c,Input)
 
diff --git a/src/global.h b/src/global.h
index a036ee5..cb2b92d 100644
--- a/src/global.h
+++ b/src/global.h
@@ -50,6 +50,7 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <stdbool.h>
+#include <glib.h>
 
 /* Forward declarations for structures the HIDs need.  */
 typedef struct BoxType BoxType, *BoxTypePtr;
diff --git a/src/mtspace.c b/src/mtspace.c
index 9d593e8..40f5413 100644
--- a/src/mtspace.c
+++ b/src/mtspace.c
@@ -55,7 +55,6 @@
 #include <dmalloc.h>
 #endif
 
-#define ABS(x) ((x) < 0 ? -(x) : (x))
 RCSID ("$Id$");
 
 /* mtspace data structures are built on r-trees. */
diff --git a/src/polygon1.c b/src/polygon1.c
index 0a3a2b5..1dba74c 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -54,9 +54,6 @@
 
 #define EPSILON (1E-8)
 #define IsZero(a, b) (fabs((a) - (b)) < EPSILON)
-#ifndef ABS
-#define ABS(x) ((x) < 0 ? -(x) : (x))
-#endif
 
 /*********************************************************************/
 /*              L o n g   V e c t o r   S t u f f                    */
